sqlwhat/check_funcs.py

```diff
@@ -10,7 +12,17 @@ def Ex(state=None):
     return Chain(state or State.root_state)
 
 def check_statement(name, index=0, missing_msg="missing statement", state=None):
-    pass
+    df = partial(dispatch, 'statement', name, slice(None))
+
+    stu_stmt_list = df(state.student_ast)
+    try: stu_stmt = stu_stmt_list[index]
+    except IndexError: state.reporter.do_test(Test(missing_msg))
+
+    sol_stmt_list = df(state.solution_ast) 
+    try: sol_stmt = sol_stmt_list[index]
+    except IndexError: raise IndexError("Can't get %s statement at index %s"%(name, index))
```

Comment:
use spaces around `%` for readability. Didn't find anything about it in PEP8 styleguide, so ignore if you disagree.